### PR TITLE
Next

### PR DESCRIPTION
**Pull request by ADmad**
This is for next major release. Notable changes:

- Core dep. bumped to 3.4.
- All deprecated code removed.
- Ability to have `Table::searchConfiguration()` method removed. I have documented how to achieve the same separation of code by adding `Table::searchManager()` instead.
- Removed `queryStringToData` option of `PrgComponent`. Setting `$this->request->data` was pretty hackish. I have documented how the search inputs can still be easily populated from query string.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,26 +87,26 @@ You can use `SearchManager::add()` method to add filter or use specific methods
...
 
-    public function searchConfiguration()
+        // Add the behaviour to your table
+        $this->addBehavior('Search.Search');
+    }
+
+    public function searchManager()
     {
-        $search = new Manager($this);
+        $searchManager = $this->behaviors()->Search->searchManager();
```
Comment (dereuromark) [inline README.md]: I believe you can also directly do 

    $this->searchManager()

This is typehinted in the IDE due to the IdeHelper and the loaded behavior :)

**Comment by dereuromark** [comment]
Looks good to be on a first glance.

**Comment by dereuromark** [comment]
Will you include https://github.com/FriendsOfCake/search/pull/167  for the release?

File: README.md (hunk 1 above)
Comment (ADmad) [inline README.md]: Calling $this->searchManager() inside Table::searchManager() will cause an infinite loop :)

Comment (dereuromark) [inline README.md]: Yeah this is a neat way of overwriting the parent "call" it seems.

Comment (dereuromark) [inline README.md]: Small downside only is that we lose some IDE clickability here probably, compared to the previous way.

**Comment by ADmad** [comment]
@burzum You too use this plugin right? Let me know if you have any feedback on this.

**Comment by burzum** [comment]
@ADmad seems to be OK for me! 👍

**Pull request merged**